queue/redis/push.go

```diff
@@ -21,10 +23,26 @@ func (c *client) Push(ctx context.Context, channel string, item []byte) error {
 		return errors.New("item is nil")
 	}
 
+	var signed []byte
+
+	var out []byte
+
+	// this should already be validated on startup
+	if c.config.SigningPrivateKey == nil || len(*c.config.SigningPrivateKey) != 64 {
+		return errors.New("no valid signing private key provided")
+	}
+
+	c.Logger.Tracef("signing item for queue %s", channel)
+
+	// sign the item using the private key generated using sign
+	//
+	// https://pkg.go.dev/golang.org/x/crypto@v0.1.0/nacl/sign
+	signed = sign.Sign(out, item, c.config.SigningPrivateKey)
```

Comment:
Interesting that the library here calls out

> Sign uses Ed25519 to sign messages. The length of messages is not hidden. Messages should be small because: 1. The whole message needs to be held in memory to be processed. 2. Using large messages pressures implementations on small machines to process plaintext without verifying the signature.

Prior to https://github.com/go-vela/server/pull/927, I would have been concerned about how long `pipeline.Build` can be. Something to keep in mind in case we wanted to add anything else to the queue... I think build, repo, and user are small enough to not cause issues. Tried digging in to _how_ long a message can be before considered to be "too long", but didn't find much. I imagine it's a pretty big number.

Reply:
good call out, i'll try out some massive payloads and see what happens. 
im not sure if the library maybe duplicates the in-memory object while processing it, but, we already store a couple pipelines in memory so im imaging it will be fine. ill test